pkg/aws/common_mutator.go

```diff
@@ -26,49 +27,65 @@ type Mutator struct {
 	Logger    micrologger.Logger
 }
 
-func MutateReleaseVersionLabel(m *Mutator, meta metav1.Object) ([]mutator.PatchOperation, error) {
+func MutateLabelFromCluster(m *Mutator, meta metav1.Object, cluster capiv1alpha2.Cluster, label string) ([]mutator.PatchOperation, error) {
 	var result []mutator.PatchOperation
 
-	if key.Release(meta) != "" {
+	if meta.GetLabels()[label] != "" {
 		return result, nil
 	}
-	// If the release version label is not set, we default it here
-	{
-		// Retrieve the Cluster ID.
-		clusterID := key.Cluster(meta)
-		if clusterID == "" {
-			return nil, microerror.Maskf(invalidConfigError, "Object has no %s label, can't detect release version.", label.Cluster)
-		}
 
-		namespace := meta.GetNamespace()
-		if namespace == "" {
-			namespace = metav1.NamespaceDefault
-		}
+	// Extract release from Cluster.
+	value := cluster.GetLabels()[label]
+	if value == "" {
+		return nil, microerror.Maskf(notFoundError, "Cluster %s did not have the label %s set.", cluster.GetName(), label)
+	}
+	m.Logger.Log("level", "debug", "message", fmt.Sprintf("Label %s is not set and will be defaulted to %s from Cluster %s.",
+		label,
+		value,
+		cluster.GetName()))
+	patch := mutator.PatchAdd(fmt.Sprintf("/metadata/labels/%s", EscapeJSONPatchString(label)), value)
+	result = append(result, patch)
+
+	return result, nil
+}
+
+func FetchCluster(m *Mutator, meta metav1.Object) (*capiv1alpha2.Cluster, error) {
```

Comment:
we move the fetching of the cluster into a separate function so it is called just once